packages/client/hmi-client/src/components/widgets/tera-input-text.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+	<div class="tera-input" :label="label" :title="title">
+		<label v-if="label" @click.self.stop="focusInput">{{ label }}</label>
+		<main :class="{ error: getErrorMessage }" @click.self.stop="focusInput">
+			<i v-if="icon" :class="icon" />
+			<input
+				ref="inputField"
+				:disabled="getDisabled"
+				:placeholder="placeholder"
+				:style="inputStyle"
+				type="text"
+				:value="displayValue()"
+				@click.stop
+				@focus="isFocused = true"
+				@focusout="$emit('focusout', $event)"
+				@blur="onBlur"
+				@input="updateValue"
+			/>
+		</main>
+	</div>
+	<aside v-if="getErrorMessage"><i class="pi pi-exclamation-circle" /> {{ getErrorMessage }}</aside>
+</template>
+
+<script setup lang="ts">
+import { numberToNist } from '@/utils/number';
+import { isNumber, isString, toNumber, isNaN } from 'lodash';
+import { CSSProperties, computed, ref } from 'vue';
+
+const props = defineProps<{
+	modelValue: string | number | undefined;
```

Comment:
For both input-text and input-number ... if the idea of the PR is to clamp down the typing, why is the `modelValue` so loosey-goosey?

Reply:
Fair, I've changed the type of `modelValue` to `number | undefined` in `tera-input-number`.